plugins/experimental/ssl_session_reuse/src/singleton.h

```diff
@@ -0,0 +1,73 @@
+/** @file
+
+  singleton.h - a containuer of connection objects
```

Comment:
As I mentioned above, there are better ways to do singletons in C++.

Reply:
Yep, let's get rid of this.